pkg/artifactory/resource/repository/repository.go

```diff
@@ -291,6 +316,14 @@ func MkResourceSchema(skeema map[string]*schema.Schema, packer packer.PackFunc,
 	}
 }
 
+func ResourceStateUpgradeV1(_ context.Context, rawState map[string]interface{}, _ interface{}) (map[string]interface{}, error) {
+	if rawState["package_type"] != "generic" {
```

Comment:
Since `MkResourceSchema` is used for all repo types, we may want to also check `rclass` attribute to only catch remote repo resources.

Reply:
Done.